nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GenerateRecord.java

```diff
@@ -0,0 +1,649 @@
+
+        final int count = recordCount.get();
+        session.adjustCounter("Records Processed", count, false);
+
+        getLogger().info("Successfully generated {} records for {}", count, flowFile);
+    }
+
+    protected Map<String, String> getFields(ProcessContext context) {
+        return context.getProperties().entrySet().stream()
+                // filter non-null dynamic properties
+                .filter(e -> e.getKey().isDynamic() && e.getValue() != null)
+                // convert to Map of user-defined field names and types
+                .collect(Collectors.toMap(
+                        e -> e.getKey().getName(),
+                        e -> context.getProperty(e.getKey()).getValue()
+                ));
+    }
+
+    private Object getFakeData(String type, Faker faker) {
+
+        // Catch these two cases ahead of calling the "discovered" Faker method below in order to return a double instead of a String
+        if (FT_LATITUDE_ALLOWABLE_VALUE_NAME.equals(type)) {
+            return Double.valueOf(faker.address().latitude());
+        }
+        if (FT_LONGITUDE_ALLOWABLE_VALUE_NAME.equals(type)) {
+            return Double.valueOf(faker.address().longitude());
+        }
+
+        // Handle Number method not discovered by programmatically getting methods from the Faker objects
+        if (FT_NUMBER.getValue().equals(type)) {
+            return faker.number().numberBetween(Integer.MIN_VALUE, Integer.MAX_VALUE);
+        }
+
+        // Handle DateAndTime methods not discovered by programmatically getting methods from the Faker objects
+        if (FT_FUTURE_DATE.getValue().equals(type)) {
+            return faker.date().future(365, TimeUnit.DAYS);
+        }
+        if (FT_PAST_DATE.getValue().equals(type)) {
+            return faker.date().past(365, TimeUnit.DAYS);
+        }
+        if (FT_BIRTHDAY.getValue().equals(type)) {
+            return faker.date().birthday();
+        }
+
+        // Handle Crypto methods not discovered by programmatically getting methods from the Faker objects
+        if (FT_MD5.getValue().equals(type)) {
+            return faker.crypto().md5();
+        }
+        if (FT_SHA1.getValue().equals(type)) {
+            return faker.crypto().sha1();
+        }
+        if (FT_SHA256.getValue().equals(type)) {
+            return faker.crypto().sha256();
+        }
+        if (FT_SHA512.getValue().equals(type)) {
+            return faker.crypto().sha512();
+        }
+
+        // If not a special circumstance, use the map to call the associated Faker method and return the value
+        try {
+            final FakerMethodHolder fakerMethodHolder = datatypeFunctionMap.get(type);
+            Object returnObject = fakerMethodHolder.getMethod().invoke(fakerMethodHolder.getMethodObject());
+            return returnObject;
+        } catch (InvocationTargetException | IllegalAccessException e) {
+            throw new ProcessException(type + " is not a valid value");
```

Comment:
Recommend including the exception:
```suggestion
            throw new ProcessException(type + " is not a valid value", e);
```